### ISSUE TITLE
Why is grid the default view?

### ISSUE THREAD

**Issue by carlmjohnson** [ogham/exa#324]
I have never wanted grid view, and I have always hated that it's the default view in `ls`. It's hard to read, and it doesn't have any advantage over a simple one column view. Sure, it can be there if someone wants it, but does it really need to be the default? Am I alone in this opinion or do other people feel like I do?

**Comment by kode9** [comment]
I agree with you, column should be default. I pretty much never use grid view and a lot of

**Comment by tqdv** [comment]
I use grid view to have a quick overview of the directory: it makes it possible to see all the files in a single screen.  
Grid view also has less information, which makes it easier to "parse" quickly.

You can always always alias exa to default to long view anyways.

**Comment by dieggsy** [comment]
I'm really not arguing one way or the other here (I don't care what the default is as long as I have aliases), but 

> Grid view also has less information, which makes it easier to "parse" quickly.

Grid view does not have less information than a _single column_ view with just a list of files, which is what @carlmjohnson originally proposed. Also, how easy it is for something to parse is less a function of how  much info there is as how it's structured IMO

> Whether fitting into a smaller vertical space is useful or not depends on how many files there are and how tall the screen is.

Right, but because of this grid view is more likely to fit more variations of those factors, in any case, which might be an argument toward it being a default (though I imagine that's _not_ an argument for you if you literally find it illegible)

In any case, it seems a lot of this comes down to personal preference. Another argument one might make is that as exa is adverstised as a 'replacement' to ls, grid view should be the default. Dunno, I think better defaults might be part of what makes improved replacements desirable in the first place. 

But again, this feels fairly low priority given the ability to change "defaults" through aliases.

**Comment by tqdv** [comment]
`exa -1` (if that's what was proposed, though I doubt it) mixes itself with the command prompt. It's too far left, and takes a lot of vertical space. Grid view leaves more of the terminal history on-screen.  
This boils down to personal preference.

I think the default view should be the one with the less information, while still being useful. So definitely not long view.

**Comment by dieggsy** [comment]
> if that's what was proposed, though I doubt it

The comment said

> and it doesn't have any advantage over a simple **one column view**

**Comment by ogham** [comment]
Fun facts: [the very first version of exa](https://the.exa.website/releases/v0.1.0), from four years ago, didn’t even have a grid view at all! Column view was the only one, and I added the grid view for compatibility with ls and because people asked for it.

> Another argument one might make is that as exa is adverstised as a 'replacement' to ls, grid view should be the default.

This is pretty much it. I don’t have to do things exactly the same way as ls, but every day I see people using grid view, and I think a tool that does something *different* would end up being seen as doing something *wrong*.

---

However, I can see the use for something like this! My own “muscle memory” is to run `ll` by default, only switching back to `ls` if I know I’m in a directory with lots of files: I’d rather scan a list downwards than scan a grid down-then-rightwards.

So how about an environment variable? Set `EXA_PREFER_COLUMNS=1` or something, and `exa` will use the columns view, `exa -g` will switch back to the grid view, and `exa -l` will be redundant.

**Comment by temp** [comment]
> So how about an environment variable? Set EXA_PREFER_COLUMNS=1 or something, and exa will use the columns view, exa -g will switch back to the grid view, and exa -l will be redundant.

Would be nice to also have env vars for other configurations. Like always showing the --icons. Or, maybe, have a .exarc for settings?

**Comment by tabby-jun** [comment]
> > So how about an environment variable? Set EXA_PREFER_COLUMNS=1 or something, and exa will use the columns view, exa -g will switch back to the grid view, and exa -l will be redundant.
> 
> Would be nice to also have env vars for other configurations. Like always showing the --icons. Or, maybe, have a .exarc for settings?

A feature that allows `exa` users to choose their own defaults will be awesome 😍

**Comment by mschilli87** [comment]
@tabby-jun:
> A feature that allows exa users to choose their own defaults will be awesome

How about adding

```sh
alias exa='exa [your defaults]'
```
to your `~/.bashrc`?

**Comment by tabby-jun** [comment]
Hi, `@mschilli87`, thanks for the workaround. It should suffice for now.
Maybe it's better to put this alias in a `.bash_aliases` file?
